src/dosext/mfs/lfn.c

```diff
@@ -1328,6 +1328,11 @@ int mfs_lfn(void)
 {
 	int carry, ret;
 
+	if (!mach_fs_enabled) {
+		CARRY;
+		return 0;
+	}
+
```

Comment:
So not QUALIFY_FILENAME per se, but its int21 equivalent [truename](http://www.delorie.com/djgpp/doc/rbinter/id/50/31.html) and so see the implementation of truename in lfn.c. You'll see it checking cds_flags for SUBST, JOIN etc, reversing CDS into a drive letter etc. Initially i started to fix the DR-DOS crash in the way you suggest, see where it uses CDS and try to handle if missing, but there were at least 5 crash cases and it seemed odd that mfs_lfn() was causing them all on non mfs drives, hence the attraction of the big switch off. Actually loss of LFN on DR-DOS < 7 actually aligned well with the fact that the same DR-DOS variants also don't support int 21/6c extended open, so LFN is pretty useless anyway.
Since as I now understand LFN is generic and not tied to mfs, should it be moved from the mfs directory?